optaplanner-core/src/main/java/org/optaplanner/core/impl/exhaustivesearch/DefaultExhaustiveSearchPhase.java

```diff
@@ -269,4 +243,35 @@ public void solvingEnded(SolverScope<Solution_> solverScope) {
         decider.solvingEnded(solverScope);
     }
 
+    public static class Builder<Solution_> extends AbstractPhase.Builder<Solution_> {
+
+        private final Comparator<ExhaustiveSearchNode> nodeComparator;
+        private final EntitySelector<Solution_> entitySelector;
+        private final ExhaustiveSearchDecider<Solution_> decider;
+
+        private boolean assertWorkingSolutionScoreFromScratch = false;
+        private boolean assertExpectedWorkingSolutionScore = false;
+
+        public Builder(int phaseIndex, String logIndentation, Termination<Solution_> phaseTermination,
+                Comparator<ExhaustiveSearchNode> nodeComparator, EntitySelector<Solution_> entitySelector,
+                ExhaustiveSearchDecider<Solution_> decider) {
+            super(phaseIndex, logIndentation, phaseTermination);
+            this.nodeComparator = nodeComparator;
+            this.entitySelector = entitySelector;
+            this.decider = decider;
+        }
+
+        public void setAssertWorkingSolutionScoreFromScratch(boolean assertWorkingSolutionScoreFromScratch) {
+            this.assertWorkingSolutionScoreFromScratch = assertWorkingSolutionScoreFromScratch;
+        }
+
+        public void setAssertExpectedWorkingSolutionScore(boolean assertExpectedWorkingSolutionScore) {
+            this.assertExpectedWorkingSolutionScore = assertExpectedWorkingSolutionScore;
+        }
+
+        @Override
+        public DefaultExhaustiveSearchPhase<Solution_> build() {
+            return new DefaultExhaustiveSearchPhase<>(this);
```

Comment:
Typically, in a builder pattern, the builder itself is not given to the thing it builds IIRC.

Reply:
An alternative to passing a builder instance is creating a constructor that accepts all possible options as parameters. In that case, this `build` method becomes yet another place where the options need to be passed to the constructor as arguments. For that reason, I prefer the way @yurloc proposes.

Reply:
Exactly as Radek says. Passing the builder instance avoids repetition. I've seen this approach in multiple sources, including the Effective Java book.